### PR TITLE
Feature/issue 708 followup change

### LINKED ISSUE #708: Insecure mode on default sync_gateway bucket
I am a lazy developer so when I start a new project I often config my iOS simulator apps to sync with `http://localhost:4985` (eg the admin port). I think this is bad practice because it leads to us wanting to do things like enable CORS on the admin port or whatever.

Instead we should have an out-of-the-box insecure mode on the regular port, maybe by adding this to the default config for the `sync_gateway` bucket:

``` js
"users": {
                "GUEST": {"disabled": false, "admin_channels": ["*"] }
            }
```

Making this change would eliminate having to configure Sync Gateway from many hello world applications and tutorials. Eg you could just start it with `./sync_gateway` and it would work for hello world.

### PR DESCRIPTION
**Pull request by tleyden**
Fixes #708

### CODE REVIEW

**Comment by tleyden** [comment]
I just realized that the warnings need to mention the name of the database.  Will push a follow up commit.

**Comment by tleyden** [comment]
Follow up commit has been pushed.

File: src/github.com/couchbase/sync_gateway/rest/server_context.go [hunk 1]
```diff
@@ -511,3 +513,59 @@ func (sc *ServerContext) Stats() map[string]interface{} {
...
+	}
+	vres, err := currentDb.Bucket.View(db.DesignDocSyncGateway, db.ViewPrincipals, viewOptions)
+	if err != nil {
+		base.Warn("Error trying to query ViewPrincipals: %v", err)
+		return []string{}
+	}
+
+	// if no users defined in config, and no users were returned from the view, add warning.
+	// NOTE: currently ignoring the fact that the config could contain only disabled=true users.
+	if len(config.Users) == 0 && len(vres.Rows) == 0 {
```
Comment (adamcfraser) [inline src/github.com/couchbase/sync_gateway/rest/server_context.go]: I think we should check config.Users first, and so avoid the view call if len(config.Users) > 0.  That would also provide a workaround if customers with tons of users see problems with the view performance - they could just stick a disabled guest user into their config.

**Comment by tleyden** [comment]
- Added test 'no guest user but one user exists in the DB'.
- Check config.Users first, and so avoid the view call if len(config.Users) > 0
- Rebased commits into single commit and force pushed.

**Comment by adamcfraser** [comment]
Looks good.

**Pull request merged**